Add Zenodo integration

davelopez opened the pull request:
Requires #18028

This PR adds a new Zenodo file source plugin based on Invenio to make the integration a bit more obvious.

This also uses new schemes for Invenio (`invenio://`) and Zenodo (`zenodo://`) file sources as they should have been from the beginning... The previous `gxfiles://` scheme in the case of existing exports will still work as expected.

![image](https://github.com/galaxyproject/galaxy/assets/46503462/2ada3994-f170-4eea-b6c2-544d22136f2e)


## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  - Set `enable_celery_tasks` to `true` in your galaxy.yml config file.
  - Add to your `file_sources_conf.yml`
    ```yml
      - type: zenodo
        id: zenodo
        doc: Zenodo is a general-purpose open-access repository developed under the European OpenAIRE program and operated by CERN. It allows researchers to deposit data sets, research software, reports, and any other research-related digital artifacts. For each submission, a persistent digital object identifier (DOI) is minted, which makes the stored items easily citeable.
        label: Zenodo
        url: https://sandbox.zenodo.org # To use the real Zenodo instance replace with https://zenodo.org
        token: ${user.user_vault.read_secret('preferences/zenodo/token')}
        # token: ${user.preferences['zenodo|token']} # Alternatively use this for retrieving the token from user preferences instead of the Vault
        public_name: ${user.preferences['zenodo|public_name']}
        writable: true
    ```
  - Add to your `user_preferences_extra_conf.yml`
    ```yml
        zenodo:
                description: Your Zenodo Integration Settings
                inputs:
                    - name: token
                      label: Personal Access Token used to create draft records and to upload files. You can manage your tokens at https://zenodo.org/account/settings/applications/
                      type: secret
                      store: vault # Requires setting up vault_config_file in your galaxy.yml
                      required: False
                    - name: public_name
                      label: Creator name to associate with new records (formatted as "Last name, First name"). If left blank "Anonymous Galaxy User" will be used. You can always change this by editing your record directly.
                      type: text
                      required: False
    ```
  - Try to export a history

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

davelopez committed:
Add Invenio scheme

Replaces the `gxfiles` scheme with `invenio` and provides backwards compatibility with existing uris.

davelopez committed:
Add Zenodo as a standalone plugin

davelopez committed:
Add getFileSourceById to fileSources composable

davelopez committed:
Add optional selected item to FilesDialog

It allows to set an initial item (file source) to browse from.

davelopez committed:
Refactor move files dialog's conversion function to utils

davelopez committed:
Support preselecting a file source in RDM export

davelopez committed:
Add Zenodo as export for histories

If the plugin is setup it will show up as a first class option when exporting a history.

davelopez committed:
Adapt choice selector in unit tests

davelopez committed:
Add unit tests for selecting specific file source

davelopez committed:
Add unit test for displaying the Zenodo tab

jmchilton reviewed (approved):
This is so ridiculously cool.

davelopez committed:
Remove info about auto-publishing

This is more accurate as the auto-publish option is not available anymore. The publishing must be done through the Zenodo website for extra security.

bgruening reviewed (approved):
Thanks for the rebase @davelopez!
